content/views.py

```diff
@@ -98,6 +103,59 @@ def current(self, request, *args, **kwargs):
         serializer = self.get_serializer(challenge)
         return Response(serializer.data)
 
+    @detail_route(methods=['get', 'post'])
+    def winner(self, request, pk=None, *args, **kwargs):
+        # winner = get_object_or_404(Challenge, pk=pk).is_user_a_winner(request.user)
+        participant = get_object_or_404(Participant, user=request.user, challenge_id=pk)
+
+        if participant is None:
+            raise NotFound("User did not participant in challenge.")
+
+        winner_status = participant.get_winning_status()
+
+        return Response(winner_status)
+
+    @list_route(methods=['get'])
+    def winning(self, request, *args, **kwargs):
+        """Returns winning status, and badge and challenge if available"""
+        # TODO: Filter by notification flag
+        participant = Participant.objects.filter(user=request.user, is_winner=True, has_been_notified=False) \
+            .order_by('date_completed') \
+            .first()
+
+        if participant is None:
+            return Response({"available": False, "badge": None, "challenge": None})
+
+        badge_settings = BadgeSettings.for_site(request.site)
+
+        if badge_settings.challenge_win is None:
+            raise NotFound('Challenge Badge not set up')
+
+        # Create badge if it doesn't already exist
+        site = Site.objects.get(is_default_site=True)
+        user_badge = award_challenge_win(site, request.user, participant)
+
+        data = OrderedDict()
+        data['available'] = user_badge is not None and participant.challenge is not None
+        data['badge'] = UserBadgeSerializer(instance=user_badge, context=self.get_serializer_context()).data
+        data['challenge'] = ChallengeSerializer(instance=participant.challenge,
+                                                context=self.get_serializer_context()).data
+        return Response(data)
+
+    @detail_route(methods=['post'])
+    def notification(self, request, pk=None, *args, **kwargs):
+        """Marks the participant as having received the winning notification"""
+        participants = Participant.objects.filter(challenge_id=pk, user=request.user, has_been_notified=False)
+
+        if participants is None:
```

Comment:
This isn't necessary. Filter would return empty and the following iteration would simply not run.